packages/amplify-util-mock/src/CFNParser/stack/index.ts

```diff
@@ -0,0 +1,275 @@
+  let result: string[] = [];
+  if (typeof node === 'string') {
+    return [];
+  }
+
+  if (isPlainObject(node) && Object.keys(node).length === 1) {
+    const fnName = Object.keys(node)[0];
+    const fnArgs = node[fnName];
+    if ('Ref' === fnName) {
+      const resourceName = fnArgs;
+      if (!Object.keys(params).includes(resourceName)) {
+        result.push(resourceName);
+      }
+    } else if ('Fn::GetAtt' === fnName) {
+      const resourceName = fnArgs[0];
+      result.push(resourceName);
+    } else if (typeof fnArgs !== 'string') {
+      for (var i = 0; i < fnArgs.length; i++) {
+        result = [...result, ...getDependencyResources(fnArgs[i], params)];
+      }
+    }
+  } else if (Array.isArray(node)) {
+    return node.reduce((acc, item) => [...acc, ...getDependencyResources(item, params)], []);
+  }
+  return result;
+}
+
+export function sortResources(resources: CloudFormationResources, params: Record<string, any>): string[] {
+  const resourceSorter: Sorter<string> = new Sorter();
+  Object.keys(resources).forEach(resourceName => {
+    const resource = resources[resourceName];
+    let dependsOn: string[] = [];
+    // intrinsic dependency
+    const intrinsicDependency = Object.values(resource.Properties)
+      .map(propValue => getDependencyResources(propValue, params))
+      .reduce((sum, val) => [...sum, ...val], []);
+
+    // Todo: enable this once e2e test invoke transformer the same way as
+    // mock
+    // throw error if one of the intrinsic resource
+    // const missingIntrinsicDeps = intrinsicDependency.filter(res => !(res in resources));
+    // if (missingIntrinsicDeps.length) {
+    //   throw new Error(
+    //     `Resource ${resourceName} has missing intrinsic dependency ${
+    //       missingIntrinsicDeps.length === 1 ? 'resource' : 'resources'
+    //     } ${missingIntrinsicDeps.join(', ')}`,
+    //   );
+    // }
+
+    if (resource.DependsOn) {
+      if (Array.isArray(resource.DependsOn) || typeof resource.DependsOn === 'string') {
+        dependsOn = typeof resource.DependsOn === 'string' ? [resource.DependsOn] : resource.DependsOn;
+        if (dependsOn.some(dependsOnResource => !(dependsOnResource in resources))) {
+          throw new Error(`Resource ${resourceName} DependsOn an non existent resource`);
+        }
+      } else {
+        throw new Error(`DependsOn block should be an array or a string for resource ${resourceName}`);
+      }
+    }
+    try {
+      resourceSorter.add(resourceName, { group: resourceName, after: [...dependsOn, ...intrinsicDependency] });
+    } catch (e) {
+      if (e.message.indexOf('Item cannot come after itself') !== -1) {
+        throw new Error(`Resource ${resourceName} can not depend on itself`);
+      }
+      if (e.message.indexOf('created a dependencies error') !== -1) {
+        throw new Error('Cyclic dependency detected in the Resources');
```

Comment:
It would help if we can print what is the dependency chain or at least the last resource name here that is taking apart in the cyclic dependency.